backend/cellguide/api/v1/upload/actions.py

```diff
@@ -0,0 +1,48 @@
+import json
+import os
+import re
+
+import requests
+from flask import jsonify, make_response
+
+from backend.common.utils.http_exceptions import ForbiddenHTTPException
+from backend.layers.auth.user_info import UserInfo
+from backend.layers.thirdparty.s3_provider import S3Provider
+
+
+def post(body: dict, token_info: dict):
+    try:
+        UserInfo(token_info).is_cxg_admin()
+    except Exception as e:
+        raise ForbiddenHTTPException("Unauthorized") from e
+
+    cell_onthology_id = body["cell_onthology_id"]
+    references = body["references"]
+    description = body["description"]
+
+    if not re.match(r"^CL_\d{7}$", cell_onthology_id):
+        raise ForbiddenHTTPException("Invalid cell_onthology_id format. Example: CL_0000030")
+
+    for reference in references:
+        validate_url(reference)
+
+    env = "dev" if os.getenv("DEPLOYMENT_STAGE") in ["rdev", "test"] else os.getenv("DEPLOYMENT_STAGE")
+
+    file_name = f"{cell_onthology_id}.json"
+    key_name = f"validated_descriptions/{file_name}"
+    bucket_name = f"cellguide-data-public-{env}"
```

Comment:
if `rdev`, we need to suffix the bucket with `env-rdev-cellguide/{stack_name}/`. otherwise, all rdevs will be writing to the dev deployment's data